lib/util/fonts/resolveFontWeight.js

```diff
@@ -1,44 +1,33 @@
-function resolveFontWeight(weight, availableWeights) {
-    weight = parseInt(weight);
+function resolveFontWeight(desiredWeight, availableWeights) {
+    desiredWeight = parseInt(desiredWeight);
 
-    if (availableWeights.indexOf(weight) !== -1) {
-        return weight;
+    if (availableWeights.indexOf(desiredWeight) !== -1) {
+        return desiredWeight;
     }
 
-    var value;
     var sortedWeights = availableWeights.slice().sort(function (a, b) { return a - b; });
 
-    if (weight === 400 && availableWeights.indexOf(500) !== -1) {
+    if (desiredWeight === 400 && availableWeights.indexOf(500) !== -1) {
         return 500;
     }
 
-    if (weight === 500 && availableWeights.indexOf(400) !== -1) {
+    if (desiredWeight === 500 && availableWeights.indexOf(400) !== -1) {
         return 400;
     }
 
-    if (weight <= 400) {
-        sortedWeights.forEach(function (possibleWeight) {
-            if (possibleWeight < 400) {
-                value = possibleWeight;
-            } else if (!value) {
-                value = possibleWeight;
-            }
-        });
+    var above = sortedWeights.filter(function (weight) { return weight > desiredWeight; });
+    var below = sortedWeights.filter(function (weight) { return weight < desiredWeight; });
+    var searchOrder;
 
-        return value;
+    if (desiredWeight > 500) {
+        // If the desired weight is greater than 500, weights above the desired weight are checked in ascending order followed by weights below the desired weight in descending order until a match is found.
+        searchOrder = above.concat(below.reverse());
+    } else {
+        // If the desired weight is less than 400, weights below the desired weight are checked in descending order followed by weights above the desired weight in ascending order until a match is found.
+        searchOrder = below.reverse().concat(above);
     }
 
-    if (weight >= 500) {
-        sortedWeights.slice().reverse().forEach(function (possibleWeight) {
-            if (possibleWeight > 500) {
-                value = possibleWeight;
-            } else if (!value) {
-                value = possibleWeight;
-            }
-        });
-
-        return value;
-    }
+    return searchOrder[0];
```

Comment:
Yeah, this is pretty close to the described algorithm. But I think the description doesn't have the luxury of already having a limited subset of available weights, so it describes checking all weights.

I thought about early returns like `lastHalf[0] || firstHalf[firstHalf.length - 1]`, but it feel less clear from reading the code what the purpose it